simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -422,24 +421,37 @@ std::vector<std::int64_t> HdMapUtils::getFollowingLanelets(
 std::vector<std::int64_t> HdMapUtils::getRoute(
   std::int64_t from_lanelet_id, std::int64_t to_lanelet_id)
 {
+  if (route_cache_.exists(from_lanelet_id, to_lanelet_id)) {
+    return route_cache_.getRoute(from_lanelet_id, to_lanelet_id);
+  }
   std::vector<std::int64_t> ret;
   const auto lanelet = lanelet_map_ptr_->laneletLayer.get(from_lanelet_id);
   const auto to_lanelet = lanelet_map_ptr_->laneletLayer.get(to_lanelet_id);
   lanelet::Optional<lanelet::routing::Route> route =
     vehicle_routing_graph_ptr_->getRoute(lanelet, to_lanelet, 0, true);
   if (!route) {
+    route_cache_.appendData(from_lanelet_id, to_lanelet_id, ret);
     return ret;
   }
   lanelet::routing::LaneletPath shortest_path = route->shortestPath();
   if (shortest_path.empty()) {
+    route_cache_.appendData(from_lanelet_id, to_lanelet_id, ret);
     return ret;
   }
   for (auto lane_itr = shortest_path.begin(); lane_itr != shortest_path.end(); lane_itr++) {
     ret.push_back(lane_itr->id());
   }
+  route_cache_.appendData(from_lanelet_id, to_lanelet_id, ret);
   return ret;
 }
 
+std::shared_ptr<traffic_simulator::math::CatmullRomSpline> HdMapUtils::getCenterPointsSpline(
+  std::int64_t lanelet_id)
+{
+  const auto center_points = getCenterPoints(lanelet_id);
+  return center_points_cache_.getCenterPointsSpline(lanelet_id);
```

Comment:
The variable center_points is not used and should not be calculated.

Reply:
> getCenterPoints(lanelet_id);
This funtion includes features caching the center points spline, but certaily, "const auto centerpoints = " is unnecessary, so delete it.

Reply:
The name get ... is inappropriate if it causes side effects.